### ISSUE TITLE
2024_06: FEATURE: ability to see CLDs and DDD stocks side-by-side

### ISSUE THREAD

**Issue by ljmoody** [lzim/teampsd#3190]
## What happens now:
- Lindsey would like the "explainer" CLDs to be able to be seen side-by-side with the stocks (what it currently floats above). When she sized to minimum window size, these were the results (not able to see/explain MM stocks hidden "under" the floating window:
- smallest size, initial pop-up location:
![image](https://github.com/lzim/teampsd/assets/124643236/7664f9db-0b65-4db3-95fb-61f2436a0e12)
- smallest size pop-up, all the way to the right:
![image](https://github.com/lzim/teampsd/assets/124643236/84ae775a-a61f-4f97-be79-6711b35c8809)
- smallest pop-up size, all the way to the left:
![image](https://github.com/lzim/teampsd/assets/124643236/254aa02a-43e7-48e7-a072-f856c00200bd)

## What is desired:
- Ability to see stocks _and_ CLD for any of the 5 modules.

**Comment by jamesmrollins** [comment]
@tejaspesquare @matomasz 

Before any work is performed on this, we need to engage with the designer to figure out if this is technically feasible with respect to the screen resolutions.  We need to be able to dynamically resize the window of the CLD without affecting the size of the CLD.  This would mean then the user would scroll around to the area of the CLD the want, without adjusting the window size.  When we get to this in the queue, I will give you a prototype to work from.

**Comment by matthewtomo** [comment]
**Discussed Dev Standup wk_3 2024_9**
@matthewtomo @tejaspesquare 

- This feature was asked about in another card (https://github.com/lzim/teampsd/issues/3207):
> Feedback Loops: scalable and movable (possibly refine?):
> - Designer believes this is not currently possible due to the scaling font size (which would shrink too small if the graphic is allowed to be scaled further down)
- Following the reccommendation of @jamesmrollins:
> Before any work is performed on this, we need to engage with the designer to figure out if this is technically feasible with respect to the screen resolutions. We need to be able to dynamically resize the window of the CLD without affecting the size of the CLD. This would mean then the user would scroll around to the area of the CLD the want, without adjusting the window size. When we get to this in the queue, I will give you a prototype to work from.
- [ ] @tejaspesquare, will check with the designer to see if we can make the window dynamically resize without resizing the diagram (which would add horizontal and vertical scrollbars for scrolling around the diagram). We are requesting just whether or not it would be feasible and a time-estimate for the work that would be required.
  - @tejaspesquare believes that if we were to make this change, then the diagram would no longer be able to scale upwards when the pop-up window is expanded. He will check with the designer to see if this is the case.
-  [ ] @jamesmrollins, in terms of a new prototype (from your previous comment), what are you expecting to provide for the Dev team and when would it be available?

**Comment by tejaspesquare** [comment]
Hello Mathew,

> > Feedback Loops: scalable and movable (possibly refine?):
> > * Designer believes this is not currently possible due to the scaling font size (which would shrink too small if the graphic is allowed to be scaled further down)
 
> * [ ]  @tejaspesquare, will check with the designer to see if we can make the window dynamically resize without resizing the diagram (which would add horizontal and vertical scrollbars for scrolling around the diagram). We are requesting just whether or not it would be feasible and a time-estimate for the work that would be required.
>   
>   * @tejaspesquare believes that if we were to make this change, then the diagram would no longer be able to scale upwards when the pop-up window is expanded. He will check with the designer to see if this is the case.

I engaged in a discussion with the designer regarding this requirement, and it became clear that managing both scrolling and resizing simultaneously is not feasible. The system can accommodate only one function at a time. Nevertheless, we can attempt to align with the requirement, although I am unable to provide an immediate estimate at this moment. 
A POC (proof of concept) will be necessary first. Presently, the designer is concentrating on the development of 508 compliance, and if time allows, they will endeavor to allocate some time to explore a potential solution.

**Comment by ljmoody** [comment]
In an effort to remove duplicate cards from our boards, I wanted to make @matthewtomo and @tejaspesquare aware of the following:

ℹ️ - similar efforts were being tracked in a duplicate card #3207 (closed today). For historical communication regarding efforts towards this feature, it may be relevant to look at that card. Especially for August 2024 comments/communications.

**Comment by matthewtomo** [comment]
**Discussed Friday Dev Standup wk_3 2024_09**
@matthewtomo @tejaspesquare 

- @ljmoody I want to clarify that the issue you closed appears to cover more behavior than just this card.
- Are we tracking that work somewhere else?

**Comment by ljmoody** [comment]
@matthewtomo - Changing the title of this card could help as this one is still hyper-specific (not the general feature development which is more along the lines of "Sim UI pop-up windows to either floating windows and/or open in a new browser tab"). The intention was to close the card(s) with specifics of which pop-ups overlay the DDD, and keep open the larger feature-related card about DDD More pop-up windows being sizable and scalable over the DDD (so both can be seen together). 
_Am I following what you meant in your first bullet? Should we live sync about this to reach a shared understanding?_

**Comment by matthewtomo** [comment]
**Discussed Monday Dev Standup wk_4 2024_09**
@matthewtomo @tejaspesquare 

- Designer will be able to work on a proof of concept for this feature after completing 508 compliance for: https://github.com/lzim/teampsd/issues/3198
- The Proof of Concept will be having a resizable window, with scrollbars to see all of the content

**Comment by matthewtomo** [comment]
> @matthewtomo - Changing the title of this card could help as this one is still hyper-specific (not the general feature development which is more along the lines of "Sim UI pop-up windows to either floating windows and/or open in a new browser tab"). The intention was to close the card(s) with specifics of which pop-ups overlay the DDD, and keep open the larger feature-related card about DDD More pop-up windows being sizable and scalable over the DDD (so both can be seen together).
> _Am I following what you meant in your first bullet? Should we live sync about this to reach a shared understanding?_

@ljmoody I think we should live sync about this

**Comment by ljmoody** [comment]
**Discussed in 9/23/2024 ad hoc meeting with @ljmoody and @matthewtomo:**
- Using 1 card to track pop-up window behavior for multiple pop-ups is too large in scope
- We believe tracking this feature (of pop-up window behavior) for each More menu item makes the best sense
- @ljmoody will create new feature cards to track development of same behavior for directives, decision meters, and episodes of care and include breadcrumb of issue cards it's related to and/or that have been closed

**Cards created:** 
#3263: Episodes of Care
#3264: Decision Meters
#3265: Directives

**Comment by matthewtomo** [comment]
**Discussed Friday Dev Standup wk_4 2024_09**
@matthewtomo @tejaspesquare 

- Work on this card will resume once Designer completes tasking on #3198 
- See above comment:

> **Discussed Monday Dev Standup wk_4 2024_09** @matthewtomo @tejaspesquare
> 
> * Designer will be able to work on a proof of concept for this feature after completing 508 compliance for: [2024_08 #story_b: DEV & TEST dynamic decision dashboard wireframe & alt text for 508 compliance #3198](https://github.com/lzim/teampsd/issues/3198)
> * The Proof of Concept will be having a resizable window, with scrollbars to see all of the content

- See also: https://github.com/lzim/teampsd/issues/3198#issuecomment-2379133077
> **Discussed Friday Dev Standup wk_4 2024_09** @matthewtomo @tejaspesquare
> 
> * Designer work is almost finished
> * Once finished, Designer will need to collaborate with Dev to implement final changes
> * Development is expected to be complete in two weeks.
>   
>   * Note, development has expanded to include more pages than the DDD based on @ljmoody's above comment
> 
> > * Re: the Sim UI Play page, this is also considered a product of the government. So yes, we need 508 compliance--and alt text.
> >   
> >   * Suggest using James' [508 legacy wireframe .pdf](https://takouba-my.sharepoint.com/:b:/g/personal/lmoody_praxisanalytix_com/EX_zRxr7nqBGiunDOWGZ6usBkxEVFLsQRMKPwStuKIOFcQ?e=f7Slrn) as an initial wireframe for this task.
> >   * If there are missing elements discovered while developing the alt text for the Sim UI Play page, please let me know and I will track those down.
> 
> ~ [#3198 (comment)](https://github.com/lzim/teampsd/issues/3198#issuecomment-2357169548)

fyi: @ljmoody
